frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-ref.ts

```diff
@@ -0,0 +1,84 @@
+import { K8sModel } from '../../api/common-types';
+import { GetAPIVersionForModel, GetGroupVersionKindForResource } from '../../api/k8s-types';
+import { K8sGroupVersionKind, K8sResourceKindReference } from '../../extensions/console-types';
+
+/**
```

Comment:
```suggestion
/**
 * @deprecated - Moving to K8sGroupVersionKind, use getGroupVersionKindForResource
 * @see getGroupVersionKindForResource
```

Reply:
these are internal utility and have not exposed any of `getReference*` utility via SDK, so not added `deprecated`